src/group_theory/abelianization.lean

```diff
@@ -170,3 +170,12 @@ lemma abelianization_congr_trans {I : Type v} [group I] (e₂ : H ≃* I) :
 mul_equiv.to_monoid_hom_injective (abelianization.hom_ext _ _ rfl)
 
 end abelianization_congr
+
+/-- The abelianization of an Abelian group is equivalent to the group itself -/
+def abelianization_of_comm_group {H : Type*} [comm_group H] :
+  abelianization H ≃* H :=
+{ to_fun := abelianization.lift (monoid_hom.id H),
+  inv_fun := abelianization.of,
+  left_inv := by {rintros ⟨a⟩, simp },
+  right_inv := by {intros a, simp },
+  map_mul' := by tidy }
```

Comment:
Given that the map `abelianization.of` always exists, I think it's nicer to define the equiv going in the other direction as an extension of `of`. Note that this means we don't have to prove `map_mul'` (even though this is trivial).
```suggestion
/-- An Abelian group is equivalent to its own abelianization. -/
@[simps] def abelianization.equiv_of_comm {H : Type*} [comm_group H] :
  H ≃* abelianization H :=
{ to_fun    := abelianization.of,
  inv_fun   := abelianization.lift (monoid_hom.id H),
  left_inv  := λ a, rfl,
  right_inv := by { rintros ⟨a⟩, refl, },
  .. abelianization.of }
```
Note also the `simps`: this automatically generates two `simp` lemmas which make this definition usable (if you are using VS code and change `simps` to `simps?`, you can see what lemmas it generates.)